src/main/java/com/mraof/minestuck/client/renderer/entity/UnderlingRenderer.java

```diff
@@ -29,13 +30,19 @@
 /**
  * A renderer that applies the correct grist texture/color to the underlings
  */
+@MethodsReturnNonnullByDefault
 public class UnderlingRenderer<T extends UnderlingEntity> extends GeoEntityRenderer<T>
 {
-	public UnderlingRenderer(EntityRendererProvider.Context context)
+	private final String underlingName;
+	
+	public UnderlingRenderer(EntityRendererProvider.Context context, String underlingName)
 	{
 		// this renderer does two simple things :
-		super(context, new UnderlingModel<>()); // render the entity with the base layer merged with a grist texture
-		this.addRenderLayer(new UnderlingDetailsLayer(this)); // render a second layer with details (eyes, mouth, etc)
+		// render the entity with the base layer merged with a grist texture
+		super(context, new DefaultedEntityGeoModel<>(Minestuck.id("underlings/" + underlingName), true));
+		// render a second layer with details (eyes, mouth, etc)
+		this.addRenderLayer(new UnderlingDetailsLayer<>(this, underlingName));
```

Comment:
Weve had an issue since our Geckolib release with the extra layers on underlings not turning red when the entity is injured. If you happen to know of a quick solution then now may be the time to implement it. Otherwise, no big deal!

Reply:
Found a fix to this. While it's very small, I think I'd prefer committing it separately. I'll add it directly to the 1.20.1 branch.